bcda/service/service_test.go

```diff
@@ -1170,6 +1178,201 @@ func (s *ServiceTestSuite) TestGetACOConfigForID() {
+	suite.Suite
+	priorityACOsEnvVar string
+	repository         *postgres.Repository
+	db                 *sql.DB
+}
+
+func TestServiceTestSuiteWithDatabase(t *testing.T) {
+	suite.Run(t, new(ServiceTestSuiteWithDatabase))
+}
+
+func (s *ServiceTestSuiteWithDatabase) SetupSuite() {
+	s.db, _ = databasetest.CreateDatabase(s.T(), "../../db/migrations/bcda/", true)
+	s.repository = postgres.NewRepository(s.db)
+}
+
+func (s *ServiceTestSuiteWithDatabase) SetupTest() {
+	s.priorityACOsEnvVar = conf.GetEnv("PRIORITY_ACO_REG_EX")
+}
+
+func (s *ServiceTestSuiteWithDatabase) TearDownTest() {
+	conf.SetEnv(s.T(), "PRIORITY_ACO_REG_EX", s.priorityACOsEnvVar)
+}
+
+// suppressions.yml
+// MBI00000001: (opted out) opted out 1 day ago and has an older record with no preference
+// MBI00000002: (opted out) opted out 10 days ago and opted in 30 days ago
+// MBI00000003: (opted out) has no previous records and opted out 1 day ago
+// MBI00000004: has a single record with no opt out preference ("")
+// MBI00000005: has no previous record but has opted in
+// MBI00000006: has previous opt out records and opted in
+// MBI00000007: has no opt out records
+// MBIs 1,2,3 should be suppressed for non TCOC requests
+func (s *ServiceTestSuiteWithDatabase) TestGetBenesByIDIntegration() {
+	cfg, err := LoadConfig()
+	if err != nil {
+		s.T().Fatal(err)
+	}
+
+	service := service{
+		repository: s.repository,
+		sp: suppressionParameters{
+			includeSuppressedBeneficiaries: false,
+			lookbackDays:                   60,
+		},
+		acoConfigs: cfg.ACOConfigs,
+	}
+
+	test_cases := []struct {
+		name      string
+		cmsID     string
+		beneCount int
+		mbis      []string
+	}{
+		{"TCOCMD request with active suppressions", "TCOCMD", 7, []string{"MBI00000001", "MBI00000002", "MBI00000003", "MBI00000004", "MBI00000005", "MBI00000006", "MBI00000007"}},
+		{"Non TCOCMD request with active suppressions", "A0001", 4, []string{"MBI00000004", "MBI00000005", "MBI00000006", "MBI00000007"}},
+	}
+
+	for _, test := range test_cases {
+		s.T().Run(test.name, func(t *testing.T) {
+			tf, err := testfixtures.New(
+				testfixtures.Database(s.db),
+				testfixtures.Dialect("postgres"),
+				testfixtures.Directory("fixtures"))
+			if err != nil {
+				assert.FailNowf(s.T(), "Failed to setup test fixtures", err.Error())
+			}
+			if err := tf.Load(); err != nil {
+				assert.FailNowf(s.T(), "Failed to load test fixtures", err.Error())
+			}
+			rc := RequestConditions{
+				CMSID: test.cmsID,
+			}
+			actualBeneCount, err := service.getBenesByFileID(context.Background(), 1, rc)
+			if err != nil {
+				s.T().Fatal(err)
+			}
+			assert.Equal(t, test.beneCount, len(actualBeneCount))
+			for i := 0; i < len(actualBeneCount); i++ {
+				assert.Equal(t, test.mbis[i], actualBeneCount[i].MBI)
+			}
+		})
+	}
+
+}
+
+func (s *ServiceTestSuiteWithDatabase) TestGetNewAndExistingBeneficiaries_RecentSinceParameterDatabaseIntegration() {
+	db := database.Connection
```

Comment:
this was an already existing test that it utilizing the database's package `init()`. I moved it under the new test suite since we should eventually remove the dependency on init() and instead use the 'CreateDatabase` function to stand up our test databases. I will continue to work on this if it fits into the sprint.

Reply:
Im not fully following all of this, maybe we can talk offline and would love to get Kevins perspective here.